### PR TITLE
Feature: Token Get NFT Info

### LINKED ISSUE #199: State-view
**Description:**
- Implement getting NFT from state and returning it as proto message

### LINKED ISSUE #200: E2E Tests
**Description:**
- Add E2E tests

### LINKED ISSUE #201: Fees
**Description:**
- Enhance the fee calculator to support the query

### PR DESCRIPTION
**Pull request by failfmi**
**Related issue(s)**:
Closes #199  
Closes #200 
Closes #201 

**Summary of the change**:
* Add `UniqueStore`: `nftExists`, `get`
* Add `hapi-fees`: `TokenGetNftInfoUsage`
* Add `StateView`: `infoForNft`
* Add `GetTokenNftInfoResourcerUsage`
* Unit Tests
* E2E tests

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

File: hapi-fees/src/main/java/com/hedera/services/usage/token/entities/NftEntitySizes.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_ENTITY_ID_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_RICH_INSTANT_SIZE;
+import static com.hederahashgraph.fee.FeeBuilder.LONG_SIZE;
+
+public enum NftEntitySizes {
+	NFT_ENTITY_SIZES;
+
+	public int fixedBytesInNftRepr() {
```
Comment (Daniel-K-Ivanov) [inline hapi-fees/src/main/java/com/hedera/services/usage/token/entities/NftEntitySizes.java:30]: Nft metadata should be added here once we update `MerkleUniqueToken` with the `metadata` property
@daniel-danev-limechain

File: hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java [hunk 2]
```diff
@@ -324,6 +332,32 @@ public StateView(
...
+			}
+			var uniqueToken = uniqueTokenStore.get(nftID);
+
+			var info = TokenNftInfo.newBuilder()
+					.setNftID(nftID)
+					.setAccountID(uniqueToken.getOwner().toGrpcAccountId())
+					.setCreationTime(Timestamp.newBuilder()
+							.setSeconds(uniqueToken.getCreationTime().getSeconds())
+							.setNanos(uniqueToken.getCreationTime().getNanos()))
+					.setMetadata(ByteString.copyFromUtf8(uniqueToken.getMemo()));
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java:349]: This should be updated once we update the `metadata` property as-well (@daniel-danev-limechain)

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Comment by Daniel-K-Ivanov** [comment]
LGTM!

File: hedera-node/src/main/java/com/hedera/services/store/tokens/unique/ExceptionalTokenUniqueStore.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import com.hederahashgraph.api.proto.java.NftID;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenCreateTransactionBody;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TokenUpdateTransactionBody;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+public enum ExceptionalTokenUniqueStore implements UniqueStore {
```
Comment (daniel-danev-limechain) [inline hedera-node/src/main/java/com/hedera/services/store/tokens/unique/ExceptionalTokenUniqueStore.java:41]: We can remove most of this code with default interface methods and then implement only what is needed

**Review by joan41868 — APPROVED** [review]

**Pull request merged**